src/package.cr

```diff
@@ -11,6 +11,8 @@ module Shards
     def initialize(@name, @resolver, @version, @is_override = false)
     end
 
+    def_equals @name, @resolver, @version
```

Comment:
This is not exactly the same comparision that was used in `commands/install.cr`

I think it works because the same resolver instance is returned if the same key/name/source is used. And there is no equality defined in Resolver, so the default reference equality is used.

Maybe the equality should be defined in Resolver and compare class, name and source.

Reply:
`Resolver#==` is already defined:
https://github.com/waj/shards/blob/rework/package/src/resolvers/resolver.cr#L24-L28

Reply:
Oh. ok then 👍